runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -250,7 +249,12 @@ State waitUntilFinish(
       }
 
       if (!hasError) {
+        // Reset the backoff, and update the new cumulative sleep time to be the remaining
+        // part of the total input sleep duration.
         backoff.reset();
+        long nanosConsumed = nanoClock.nanoTime() - startNanos;
+        Duration consumed = Duration.millis((nanosConsumed + 999999) / 1000000);
+        backoff = backoff.withMaxCumulativeBackoff(duration.minus(consumed));
```

Comment:
Could we track the cumulative time inside the FlexibleBackoff, and have a getCumulativeTime() or getRemainingTime()?

I think that will be more cleaner on the client side, and it also can be reused.